algebird-core/src/main/scala/com/twitter/algebird/MomentsGroup.scala

```diff
@@ -102,6 +146,96 @@ sealed class Moments(val m0D: Double, val m1: Double, val m2: Double, val m3: Do
 }
 
 object Moments {
+  final class MomentsState(
+      var count: Double,
+      var mean: Double,
+      var m2: Double,
+      var m3: Double,
+      var m4: Double
+  ) {
+
+    def +=(b: Moments): this.type = {
+      /*
+       * Unfortunately we copy the code from the monoid's plus implementation,
+       * but we do it to avoid allocating a new Moments on every item in the
+       * loop. the Monoid laws test that sum matches looping on plus
+       */
+      val countCombined = count + b.m0D
```

Comment:
I guess we could use a variant of Kahan summation on countCombined, right

Reply:
Yup, and if we had the fold version, or `KahanState`, it would fit nicely here.